cardano-testnet/test/Spec/ShutdownOnSlotSynced.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Spec.ShutdownOnSlotSynced
+  ( hprop_shutdownOnSlotSynced
+  ) where
+
+import           Control.Monad
+import           Data.Function
+import           Data.Int
+import           Data.Maybe
+import           Data.Ord
+import           GHC.Num
+import           Hedgehog (Property, assert, (===))
+import           Text.Show (Show (..))
+
+import           Data.Bool ((&&))
+import           Data.Either (Either (Right), isRight)
+import           Data.List (filter, isInfixOf, last, lines, words)
+import           GHC.IO.Exception (ExitCode (ExitSuccess))
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.File as H
+import qualified Hedgehog.Extras.Test.Process as H
+import qualified Test.Base as H
+import           Testnet.Cardano (TestnetNode (..), TestnetNodeOptions (TestnetNodeOptions),
+                   TestnetOptions (..), TestnetRuntime (..), defaultTestnetOptions, testnet)
+import qualified Testnet.Cardano as TC
+import qualified Testnet.Conf as H
+import           Text.Read (read)
+
+hprop_shutdownOnSlotSynced :: Property
+hprop_shutdownOnSlotSynced = H.integration . H.runFinallies . H.workspace "chairman" $ \tempAbsBasePath' -> do
+  -- Start a local test net
+  conf <- H.noteShowM $ H.mkConf tempAbsBasePath' Nothing
+  let maxSlot = 500
+  let fastTestnetOptions = defaultTestnetOptions
+        { epochLength = 300
+        , slotLength = 0.01
+        , bftNodeOptions =
+          [ TestnetNodeOptions
+              { extraNodeCliArgs = ["--shutdown-on-slot-synced", show maxSlot]
+              }
+          ]
+        }
+  TC.TestnetRuntime { bftNodes = [TestnetNode {..}] } <- testnet fastTestnetOptions conf
+
+  -- Wait for the node to exit
+  mExitCodeRunning <- H.waitSecondsForProcess (80 :: Int) nodeProcessHandle
+
+  -- Check results
+  when (isRight mExitCodeRunning) $ do
+    H.cat nodeStdout
+    H.cat nodeStderr
+  mExitCodeRunning === Right ExitSuccess
+  log <- H.readFile nodeStdout
+  slotTip <- H.noteShow
+              $ read @Integer
```

Comment:
Let's also catch this potential exception here and fail properly. `*** Exception: Prelude.read: no parse` may potentially be a nightmare to debug.

Reply:
I'm not sure what the "proper" way of handling this is, but I'll use `fail` and explicit error messages.